File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int n=nums.size();
        int i=0,sum=0;
        int length=n+1;
        for(int j=0;j<n;j++){
            sum+=nums[j];
            while(sum>=target){
                length=min(length,j-i+1);
                sum-=nums[i];
                i++;
            }
        }
        if(length!=n+1){
            return length;
        }else{
            return 0;
        }
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        n=len(nums)
        length=n+1
        j=0
        sum=0
        for i in range(n):
            sum+=nums[i]
            while(sum>=target):
                length=min(length,i-j+1)
                sum-=nums[j]
                j+=1
        if(length!=n+1):
            return length
        else:
            return 0